File: LC 899.cpp
class Solution {
public:
    string orderlyQueue(string s, int k) {
        if (k >= 2) {
            sort(s.begin(), s.end());
            return s;
        }
        string ans;
        char minChar = 'z';
        int n = s.length();
        for (int i = 0; i < n; ++i) {
            if (s[i] <= minChar) {
                string tmp = s.substr(i) + s.substr(0, i);
                if (ans == "" || ans > tmp) {
                    ans = tmp;
                }
            }
        }
        return ans;
    }
};

File: 1000/LC 899.go
func orderlyQueue(s string, k int) string {
    if k >= 2 {
        a := []byte(s)
        sort.Slice(a, func(i, j int) bool {
            return a[i] < a[j]
        })
        return string(a)
    }

    ans, mn := s, 'z'
    for i, c := range s {
        if c < mn {
            mn = c
            ans = s[i:] + s[:i] 
        } else if c == mn {
            tmp := s[i:] + s[:i]
            if tmp < ans {
                ans = tmp
            }
        }
    } 

    return ans
}